client/question/services/FeedbackGeneratorService.js

```diff
@@ -268,28 +360,45 @@ tie.factory('FeedbackGeneratorService', [
      * @private
      */
     var _getCorrectnessTestFeedback = function(
-      outputFunctionName, testCase, observedOutput) {
+      outputFunctionName, testCase, observedOutput, testSuiteId,
+      testSuiteIndex) {
       var allowedOutputExample = testCase.getAnyAllowedOutput();
       var feedback = FeedbackObjectFactory.create(
         FEEDBACK_CATEGORIES.INCORRECT_OUTPUT_FAILURE);
-      // TODO(eyurko): Add varied statements for when code is incorrect.
-      feedback.appendTextParagraph('Your code produced the following result:');
-      if (outputFunctionName) {
-        feedback.appendCodeParagraph(
-          'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
-          ('Output (after running ' + outputFunctionName + '): ' +
-            _jsToHumanReadable(observedOutput)));
-        feedback.appendTextParagraph(
-          'However, the expected output of ' + outputFunctionName + ' is:');
+      if (testSuiteIndex == lastTestSuiteIndex) {
+        // Same test suite; attempt number > 1
+        attemptCounter[testSuiteId] += 1;
+        if (attemptCounter[testSuiteId] == 2 && testSuiteId != 'SAMPLE_INPUT') {
+          // Display expected output
+          feedback.appendTextParagraph(
+            _selectCorrectnessFeedback(correctnessFeedbackText,
+            'expected_output'));
+          feedback.appendCodeParagraph(
+            'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
+            'Expected Output: ' + _jsToHumanReadable(allowedOutputExample));
+          return feedback;
+        }
       } else {
-        feedback.appendCodeParagraph(
-          'Input: ' + _jsToHumanReadable(testCase.getInput()) + '\n' +
-          'Output: ' + _jsToHumanReadable(observedOutput));
-        feedback.appendTextParagraph('However, the expected output is:');
+        // New / next test suite
+        lastTestSuiteIndex = testSuiteIndex;
+        if (testSuiteId != 'SAMPLE_INPUT') {
+          // Display input user should try
+          feedback.appendTextParagraph(
+            _selectCorrectnessFeedback(correctnessFeedbackText,
+            'input_to_try'));
+          feedback.appendCodeParagraph(
+            'Input: ' + _jsToHumanReadable(testCase.getInput()));
+          attemptCounter[testSuiteId] = 1;
+          return feedback;
+        }
       }
-      feedback.appendCodeParagraph(
-        _jsToHumanReadable(allowedOutputExample));
-      feedback.appendTextParagraph('Could you fix this?');
+      // Allow user to display output
```

Comment:
Ah, I thought we were only going to do this after they've passed a "only sample input" and "only sample input + expected output" stage? Otherwise I suspect it'll be too much of a crutch....

EDIT: sorry, misread, I missed that the two previous clauses actually returned stuff. I think if you get rid of the SAMPLE_INPUT special-casing then you can put the stuff below in an else branch to more clearly indicate it's an alternative to the previous two scenarios, rather than an add-on.